ts/features/bonus/cgn/components/detail/__test__/CgnInfoboxDetail.test.tsx

```diff
@@ -0,0 +1,91 @@
+import { render } from "@testing-library/react-native";
+import * as React from "react";
+import { Card } from "../../../../../../../definitions/cgn/Card";
+import CgnInfoboxDetail from "../CgnInfoboxDetail";
+import { StatusEnum as CgnActivatedStatusEnum } from "../../../../../../../definitions/cgn/CardActivated";
+import TypedI18n from "../../../../../../i18n";
+import { localeDateFormat } from "../../../../../../utils/locale";
+import {
+  CardRevoked,
+  StatusEnum as CgnRevokedStatusEnum
+} from "../../../../../../../definitions/cgn/CardRevoked";
+import { StatusEnum as CgnExpiredStatusEnum } from "../../../../../../../definitions/cgn/CardExpired";
+import { StatusEnum as CgnPendingStatusEnum } from "../../../../../../../definitions/cgn/CardPending";
+
+const cgnStatusActivated: Card = {
+  status: CgnActivatedStatusEnum.ACTIVATED,
+  activation_date: new Date("2020-03-04"),
+  expiration_date: new Date("2037-02-20")
+};
+
+const cgnStatusRevoked: Card = {
+  status: CgnRevokedStatusEnum.REVOKED,
+  revocation_date: new Date("2030-02-20"),
+  activation_date: new Date("2020-03-04"),
+  expiration_date: new Date("2037-02-20"),
+  revocation_reason: "A reason to revoke" as CardRevoked["revocation_reason"]
+};
+
+const cgnStatusExpired: Card = {
+  status: CgnExpiredStatusEnum.EXPIRED,
+  activation_date: new Date("2020-03-04"),
+  expiration_date: new Date("2037-02-20")
+};
+
+const cgnStatusPending: Card = {
+  status: CgnPendingStatusEnum.PENDING
+};
+
+describe("CgnInfoboxDetail", () => {
+  it("Activated status", () => {
+    const component = render(
+      <CgnInfoboxDetail cgnDetail={cgnStatusActivated} />
+    );
+    const activatedLabel = TypedI18n.t("bonus.cgn.detail.information.active", {
+      date: localeDateFormat(
+        cgnStatusActivated.expiration_date,
+        TypedI18n.t("global.dateFormats.shortFormat")
+      )
+    });
+    expect(component).not.toBeNull();
+    expect(component.queryByTestId("infobox-text")).toHaveTextContent(
+      activatedLabel
+    );
+  });
+
+  it("Revoked status", () => {
+    const component = render(<CgnInfoboxDetail cgnDetail={cgnStatusRevoked} />);
+    const revokedLabel = TypedI18n.t("bonus.cgn.detail.information.revoked", {
+      reason: cgnStatusRevoked.revocation_reason
+    });
+
+    expect(component).not.toBeNull();
+    expect(component.queryByTestId("infobox-text-warning")).toBeDefined();
```

Comment:
```suggestion
    expect(component.queryByTestId("infobox-text-warning")).toBeTruthy();
```

The condition `toBeDefined` is `true` also when the query is not satisfied. For example:

```typescript
 expect(component.queryByTestId("infobox-that-doesnt-exists")).toBeDefined();
```
doesn't fail.

Reply:
Thanks for the hint! I was trying to add one some more tests cases, I'm going to add through next Commit